samlsp/middleware.go

```diff
@@ -254,6 +257,40 @@ func (m *Middleware) Authorize(w http.ResponseWriter, r *http.Request, assertion
 	http.Redirect(w, r, redirectURI, http.StatusFound)
 }
 
+func (m *Middleware) GetRedirectURI(r *http.Request, assertion *saml.Assertion) (string, error) {
+	secretBlock := x509.MarshalPKCS1PrivateKey(m.ServiceProvider.Key)
+
+	redirectURI := "/"
+	if relayState := r.Form.Get("RelayState"); relayState != "" {
```

Comment:
Is it safe to assume that `r.ParseForm()` would always be called before `GetRedirectURI`? We should at least add a comment about this assumption.

Reply:
ParseForm is idempotent, so probably best to call it again. Will fix for the sake of possible future tears.